common-rest/src/main/java/org/uniprot/api/rest/openapi/ApiDocConfig.java

```diff
@@ -0,0 +1,188 @@
+
+    private static final List<String> STRING_SERILISER_LIST =
+            List.of(
+                    "#/components/schemas/UniProtKBId",
+                    "#/components/schemas/ProteomeId",
+                    "#/components/schemas/Author",
+                    "#/components/schemas/Journal",
+                    "#/components/schemas/LocalDate",
+                    "#/components/schemas/Locator",
+                    "#/components/schemas/PublicationDate",
+                    "#/components/schemas/UniParcId",
+                    "#/components/schemas/ECNumber",
+                    "#/components/schemas/FeatureDescription",
+                    "#/components/schemas/FeatureId",
+                    "#/components/schemas/Flag",
+                    "#/components/schemas/IsoformId",
+                    "#/components/schemas/UniProtKBDatabase",
+                    "#/components/schemas/UniProtKBAccession",
+                    "#/components/schemas/UniRefEntryId",
+                    "#/components/schemas/UniRuleId");
+
+    private static final Comparator<Parameter> parameterComparator =
+            Comparator.comparing(Parameter::getRequired)
+                    .reversed()
+                    .thenComparing(Parameter::getName, Comparator.comparing("size"::equals))
+                    .thenComparing(Parameter::getName, Comparator.comparing("download"::equals));
+
+    @Bean
+    SpringDocConfigProperties springDocConfigProperties() {
+        return new SpringDocConfigProperties();
+    }
+
+    @Bean
+    ObjectMapperProvider objectMapperProvider(SpringDocConfigProperties springDocConfigProperties) {
+        ObjectMapperProvider mapperProvider = new ObjectMapperProvider(springDocConfigProperties);
+        ObjectMapper openApiMapper = mapperProvider.jsonMapper();
+        openApiMapper.setAnnotationIntrospector(new ApiDocsAnnotationIntrospector());
+        return mapperProvider;
+    }
+
+    @Bean
+    public OpenApiCustomiser defaultOpenApiCustomizer(ObjectMapperProvider objectMapperProvider) {
+        return openApi -> {
+            customizeSchema(openApi);
+            sortRequestParametersForGetAndPostMethods(openApi.getPaths().values());
+        };
+    }
+
+    private static void sortRequestParametersForGetAndPostMethods(Collection<PathItem> pathItems) {
+        for (PathItem item : pathItems) {
+            if (hasParameterToSort(item.getGet())) {
+                item.getGet().getParameters().sort(parameterComparator);
+            }
+            if (hasParameterToSort(item.getPost())) {
+                item.getPost().getParameters().sort(parameterComparator);
+            }
+        }
+    }
+
+    private static boolean hasParameterToSort(Operation operation) {
+        return operation != null && notNullNotEmpty(operation.getParameters());
+    }
+
+    private void customizeSchema(OpenAPI openApi) {
+        Map<String, Schema> schemaMap = openApi.getComponents().getSchemas();
```

Comment:
The reason is the Schema class is defined with Generics "Schema<T>" in spring-docs code, and getSchemas() (code from spring-docs) method does not return Schema with a Generic Type and There is nothing that we can do in our code to avoid this warning.
Sonar does not fail. I executed sonar locally and it passed.